beacon_node/http_api/src/lib.rs

```diff
@@ -3768,26 +3768,42 @@ pub fn serve<T: BeaconChainTypes>(
+            |committee_subscriptions: Vec<api_types::BeaconCommitteeSubscription>,
              validator_subscription_tx: Sender<ValidatorSubscriptionMessage>,
              task_spawner: TaskSpawner<T::EthSpec>,
              chain: Arc<BeaconChain<T>>| {
                 task_spawner.blocking_json_task(Priority::P0, move || {
-                    let subscriptions: std::collections::BTreeSet<_> = subscriptions
-                        .iter()
-                        .map(|subscription| {
-                            chain
-                                .validator_monitor
-                                .write()
-                                .auto_register_local_validator(subscription.validator_index);
-                            api_types::ValidatorSubscription {
-                                attestation_committee_index: subscription.committee_index,
-                                slot: subscription.slot,
-                                committee_count_at_slot: subscription.committees_at_slot,
-                                is_aggregator: subscription.is_aggregator,
-                            }
-                        })
-                        .collect();
+                    let mut subscriptions: std::collections::BTreeSet<_> = Default::default();
+                    let mut registrations = Vec::new();
+                    let (finalized_beacon_state, _, _) =
+                        StateId(CoreStateId::Finalized).state(&chain)?;
+                    for subscription in committee_subscriptions.iter() {
+                        chain
+                            .validator_monitor
+                            .write()
+                            .auto_register_local_validator(subscription.validator_index);
+                        // Register the validator with the `CustodyContext`
+                        if let Ok(effective_balance) = finalized_beacon_state
+                            .get_effective_balance(subscription.validator_index as usize)
+                        {
+                            registrations
+                                .push((subscription.validator_index as usize, effective_balance));
+                        }
+                        subscriptions.insert(api_types::ValidatorSubscription {
+                            attestation_committee_index: subscription.committee_index,
+                            slot: subscription.slot,
+                            committee_count_at_slot: subscription.committees_at_slot,
+                            is_aggregator: subscription.is_aggregator,
+                        });
+                    }
+                    chain
+                        .data_availability_checker
+                        .custody_context()
+                        .register_validators::<T::EthSpec>(
+                            registrations,
+                            chain.slot().unwrap(),
+                            &chain.spec,
+                        );
```

Comment:
In my POC I also used the builder registration, which is a much faster signal for those that use it

Reply:
Might also make sense to use the `prepare_beacon_proposer` endpoint as that works for nodes not running mev workflows too.